### PR TITLE
Require label for autoscaling to be enabled

### PR DESCRIPTION
**Pull request by sharnoff**
Adds the `autoscaling.neon.tech/enabled` label and requires that it's present for the autoscaler-agent to run.

### CODE REVIEW

**Comment by sharnoff** [comment]
Current plan: wait until control plane is next released, then merge this, release & deploy. Because it's a breaking change, probably bump to v0.2

File: pkg/api/vminfo.go
Comment (tychoish) [inline pkg/api/vminfo.go:21]: is this a request or just producing a copy of the object. If it's a request, I'd be inclined to combine it with the pod.Ready op. 

also I think this doesn't need to be a generic and can just accept the interface as the type of the argument? I believe that go's calling convention allows conversions.

File: pkg/plugin/trans.go
Comment (tychoish) [inline pkg/plugin/trans.go:221]: I don't think there's anything gained by declaring `verdict` here

File: pkg/plugin/trans.go
Comment (tychoish) [inline pkg/plugin/trans.go:234]: I'd avoid the concat in the fmtstring at least, and probably also avoid not just returning the result of the `fmt.Sprintf` directly.

File: pkg/api/vminfo.go
Comment (sharnoff) [reply pkg/api/vminfo.go:21]: Just produces a copy of the object, should be ok :)

Removed the generic.

**Review by tychoish — APPROVED** [review]

File: pkg/plugin/trans.go
Comment (sharnoff) [reply pkg/plugin/trans.go:222]: Leaving as-is, to match the rest of the file. May change in a follow-up PR, but undecided.

File: pkg/plugin/trans.go
Comment (sharnoff) [reply pkg/plugin/trans.go:235]: Leaving as-is, to match the rest of the file. May change in a follow-up PR, but undecided.

**Pull request merged**
